src/main/java/com/ldtteam/common/network/AbstractServerPlayMessage.java

```diff
@@ -0,0 +1,16 @@
+package com.ldtteam.common.network;
+
+import net.minecraft.server.level.ServerPlayer;
+import net.neoforged.fml.LogicalSide;
+
+/**
+ * Client (sender) -> Server (receiver) message
+ */
+public abstract class AbstractServerPlayMessage extends AbstractPlayMessage<ServerPlayer> implements IServerboundDistributor
+{
+    @Override
+    protected LogicalSide getExecutionSide()
+    {
+        return LogicalSide.SERVER;
+    }
+}
```

Comment:
Instead of the generic `extends Player` stuff, why not have the base class only deal in `Player`, and have this class have a final override for `Player` that calls an abstract `ServerPlayer` method overload?  That way the derived classes can still have `ServerPlayer` and they don't need to have the `isLogicalServer` parameter because it's redundant.